dds/DCPS/DataReaderImpl.cpp

```diff
@@ -2648,6 +2648,7 @@ DataReaderImpl::ownership_filter_instance(const SubscriptionInstance_rch& instan
 #ifndef OPENDDS_NO_OWNERSHIP_KIND_EXCLUSIVE
   if (this->is_exclusive_ownership_) {
 
+    ACE_Guard<ACE_RW_Thread_Mutex> write_guard(writers_lock_);
```

Comment:
Other uses of this lock use ACE_READ_GUARD / ACE_WRITE_GUARD.  I think we should stick with that for consistency and clarity when using the RW flavor of mutex.